### PR TITLE
Implement check_sig etc.

### PR DESCRIPTION
**Pull request by Weber-H2O2**
WIP now

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -212,52 +213,38 @@ pub async fn poll_for_p2p(
...
+    for i in 0..32 {
+        signature_a[i + 32] = s_vec[i];
+    }
+    let signature = Signature::parse(&signature_a);
 
-    let is_correct = SECP256K1.verify(&msg, &secp_sig, &pk).is_ok();
+    let message = Message::parse(&BigInt::to_vec(msg).try_into().unwrap());
 
-    println!("public key: {:}", pk);
-    println!("address: {:}", public_key_address(&pk));
```
Comment (eigmax) [inline src/common.rs:249]: We may console_log the address for debugging

File: src/common.rs [hunk 2]
```diff
@@ -212,52 +213,38 @@ pub async fn poll_for_p2p(
...
+    let pk_vec = BigInt::to_vec(&pk.bytes_compressed_to_big_int());
 
+    let mut pubkey_a = [0u8; 65];
+    for i in 0..65 {
+        pubkey_a[i] = pk_vec[i];
+    }
+    let pubkey = PublicKey::parse(&pubkey_a).unwrap();
 
-    assert!(is_correct);
+    assert!(verify(&message, &signature, &pubkey));
```
Comment (eigmax) [inline src/common.rs]: don't assert here, return the error.

Comment (eigmax) [reply src/common.rs]: where is the verify from?

Comment (eigmax) [reply src/common.rs]: check out the example here:  https://github.com/paritytech/libsecp256k1/blob/v0.3.5/tests/verify.rs#L57

Comment (Weber-H2O2) [reply src/common.rs]: OK.

File: src/common.rs [hunk 3]
```diff
@@ -235,8 +238,11 @@ pub fn check_sig(r: &Scalar, s: &Scalar, msg: &BigInt, pk: &Point) -> bool {
     let pubkey_a = pk.get_element().serialize();
 
     let pubkey = secp256k1::PublicKey::parse(&pubkey_a).unwrap();
+
+    #[cfg(target_arch = "wasm32")]
     crate::console_log!("pubkey: {:?}", pubkey);
-    crate::console_log!("address: {:?}", public_key_address(&pubkey));
+    #[cfg(target_arch = "wasm32")]
+    crate::console_log!("address: {:?}", hex::encode(public_key_address(&pubkey)));
```
Comment (eigmax) [inline src/common.rs:245]: public_key_address does not comply with https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md, plz fix it later in another PR. @Weber-H2O2

**Review by eigmax — APPROVED** [review]

**Pull request merged**